src/pivot-table/hooks/use-column-width.ts

```diff
@@ -96,57 +99,79 @@ export default function useColumnWidth(
   });
 
   /**
-   * The widths of the left columns. Scales the width to fit LEFT_SIDE_MAX_WIDTH_RATIO * rect.width if wider than that
+   * Calculates widths of the left columns as well as the sum of the widths.
+   * If the sum of widths exceed LEFT_SIDE_MAX_WIDTH_RATIO * rect.width, the left side will become scrollable
    */
-  const leftGridWidthInfo = useMemo<LeftGridWidthInfo>(() => {
-    const getColumnWidth = (columnWidth: ColumnWidth | undefined, fitToContentWidth: number) => {
-      switch (columnWidth?.type) {
-        case ColumnWidthType.Pixels:
-          return getPixelValue(columnWidth.pixels);
-        case ColumnWidthType.Percentage:
-          return getPercentageValue(columnWidth.percentage) * rect.width;
-        default:
-          // fit to content / auto
-          return fitToContentWidth;
-      }
-    };
-
-    let sumOfWidths = 0;
+  const calculateLeftGridWidthInfo = useCallback(
+    (widthOverride?: number, overrideIndex?: number) => {
+      const getColumnWidth = (columnWidth: ColumnWidth | undefined, fitToContentWidth: number) => {
+        switch (columnWidth?.type) {
+          case ColumnWidthType.Pixels:
+            return getPixelValue(columnWidth.pixels);
+          case ColumnWidthType.Percentage:
+            return getPercentageValue(columnWidth.percentage) * rect.width;
+          default:
+            // fit to content / auto
+            return fitToContentWidth;
+        }
+      };
 
-    const lastRow = headersData.data[headersData.size.y - 1] as HeaderCell[];
-    const columnWidths = lastRow.map((cell, index) => {
-      let width: number;
+      let sumOfWidths = 0;
+
+      const lastRow = headersData.data[headersData.size.y - 1] as HeaderCell[];
+      const columnWidths = lastRow.map((cell, index) => {
+        let width: number;
+
+        if (widthOverride && overrideIndex !== undefined && overrideIndex === index) {
```

Comment:
```suggestion
        if (widthOverride && overrideIndex && overrideIndex === index) {
```

Reply:
Don't think that would work if `overrideIndex`is `0`

Reply:
override index can be 0